File: frontend/src/types/invoice.ts
export type InvoiceStatus = "Pending" | "Paid" | "Cancelled";
export type PaymentMethod = "Cash" | "UPI" | "Card" | "Bank Transfer";

export interface InvoiceItem {
  productId?: string;
  product_id?: string;
  product?: string;
  quantity: number;
  price: number;
}

export interface Invoice {
  id?: string;
  _id?: string;
  customer: string;
  customer_id?: string;
  items?: InvoiceItem[];
  total: number;
  status?: InvoiceStatus;
  payment_method?: PaymentMethod;
  created_at?: string;
  date?: string;
  razorpay_order_id?: string;
  razorpay_payment_id?: string;
  payment_gateway?: string;
  payment_status?: string;
}

File: backend/app/schemas/invoice.py
from typing import List, Literal, Optional

from pydantic import BaseModel, Field


InvoiceStatus = Literal["Pending", "Paid", "Cancelled"]
PaymentMethod = Literal["Cash", "UPI", "Card", "Bank Transfer"]


class InvoiceItem(BaseModel):
    product: str
    product_id: Optional[str] = None
    quantity: int = Field(gt=0)
    price: float = Field(ge=0)


class InvoiceCreate(BaseModel):
    customer: str
    customer_id: Optional[str] = None
    items: List[InvoiceItem] = Field(min_length=1)
    total: float = Field(ge=0)
    status: InvoiceStatus = "Paid"
    payment_method: PaymentMethod = "Cash"
